python/ray/train/collective/__init__.py

```diff
@@ -0,0 +1,14 @@
+from ray.train.v2._internal.constants import is_v2_enabled
+
+if is_v2_enabled():
```

Comment:
I forget if we discussed this already -- do we need to gate this?

Reply:
Raising ImportError now:

```
(ray4) tseah@tseah-LV3607J62K ray % RAY_DEDUP_LOGS=0 python ../playground/dummy_train.py
Traceback (most recent call last):
  File "/Users/tseah/ray/../playground/dummy_train.py", line 5, in <module>
    import ray.train.collective
  File "/Users/tseah/ray/python/ray/train/collective/__init__.py", line 14, in <module>
    raise ImportError(
ImportError: `ray.train.collective` is only available in Ray Train v2. To enable it, please set `RAY_TRAIN_V2_ENABLED=1`.
```

Still works with RAY_TRAIN_V2_ENABLED=1